test/integration/consul-container/test/envoy_extensions/wasm_test.go

```diff
@@ -0,0 +1,626 @@
+							},
+						},
+						"Configuration": "plugin configuration",
+					},
+				},
+			},
+		}},
+	}
+
+	_, _, err = consul.ConfigEntries().Set(&defaults, nil)
+	require.NoError(t, err, "could not set config entries")
+
+	// Check that header is present after wasm applied
+	c2 := cleanhttp.DefaultClient()
+
+	// The wasm plugin is not always applied on the first call. Retry and see if it is loaded.
+	retryStrategy := func() *retry.Timer {
+		return &retry.Timer{Timeout: 5 * time.Second, Wait: time.Second}
+	}
+	retry.RunWith(retryStrategy(), t, func(r *retry.R) {
+		res2, err := c2.Get(fmt.Sprintf("http://localhost:%d", port))
+		require.NoError(r, err)
+
+		if value := res2.Header.Get(http.CanonicalHeaderKey("x-test")); value == "" {
+			r.Fatal("test header missing after WASM applied")
+		}
+	})
+}
+
+// TestWASMLocal Summary
+// This test ensures that a WASM extension with basic functionality is executed correctly.
+// The extension takes an incoming request and adds the header "x-test:true"
+func TestWASMLocal(t *testing.T) {
+	t.Parallel()
+
+	cwd, err := os.Getwd()
+	require.NoError(t, err, "could not get current working directory")
+	hostWASMDir := fmt.Sprintf("%s/testdata/wasm_test_files", cwd)
+
+	buildWASM(t, hostWASMDir)
+
+	cluster, _, _ := topology.NewCluster(t, &topology.ClusterConfig{
+		NumServers:                1,
+		NumClients:                1,
+		ApplyDefaultProxySettings: true,
+		BuildOpts: &libcluster.BuildOptions{
+			Datacenter:             "dc1",
+			InjectAutoEncryption:   true,
+			InjectGossipEncryption: true,
+		},
+	})
+
+	clientService, _ := createServices(t, cluster)
+	_, port := clientService.GetAddr()
+	_, adminPort := clientService.GetAdminAddr()
+
+	libassert.AssertUpstreamEndpointStatus(t, adminPort, "static-server.default", "HEALTHY", 1)
+	libassert.GetEnvoyListenerTCPFilters(t, adminPort)
+
+	libassert.AssertContainerState(t, clientService, "running")
+	libassert.AssertFortioName(t, fmt.Sprintf("http://localhost:%d", port), "static-server", "")
+
+	// Check header not present
+	c1 := cleanhttp.DefaultClient()
+	res, err := c1.Get(fmt.Sprintf("http://localhost:%d", port))
+	require.NoError(t, err)
+
+	// check that header DOES NOT exist before wasm applied
+	if value := res.Header.Get(http.CanonicalHeaderKey("x-test")); value != "" {
+		t.Fatal("unexpected test header present before WASM applied")
+	}
+
+	// wire up the wasm filter
+	consul := cluster.APIClient(0)
+	defaults := api.ServiceConfigEntry{
+		Kind:     api.ServiceDefaults,
+		Name:     "static-server",
+		Protocol: "http",
+		EnvoyExtensions: []api.EnvoyExtension{{
+			Name: "builtin/wasm",
+			Arguments: map[string]any{
+				"Protocol":     "http",
+				"ListenerType": "inbound",
+				"PluginConfig": map[string]any{
+					"VmConfig": map[string]any{
+						"Code": map[string]any{
+							"Local": map[string]any{
+								"Filename": "/wasm_add_header.wasm",
+							},
+						},
+						"Configuration": "plugin configuration",
```

Comment:
The `Configuration` field is optional. I think you can remove it since your Wasm plugin doesn't need any config. If you remove it and things break, it is a bug in the Wasm extension ;)